### DISCUSSION TITLE
Provide builtin read only file to web worker

### DISCUSSION THREAD

**Discussion by dimitrisfasoulas** [eclipse-langium/langium#1396] [Q&A]
Hello. First of all, thank you for the wonderful job done with Langium. It is a joy to work with.

I have a question regarding providing a built-in library file to Monaco through the web worker.

In the [BuiltIn Libraries guide](https://langium.org/guides/builtin-library/) there is a clear example of how to achieve this.

However, the last part that provides the file through a custom FileSystemProvider is VsCode specific and will not work with Monaco on the web.

As a result although I can use the build in library on Monaco and its exported nodes appear for autocompletion, peek definitions or go to references shows an empty file  (although it has a correct  references number and line numbers)
![image](https://github.com/eclipse-langium/langium/assets/11027314/6102270e-7e35-42d3-8f14-f34a55e97793)
 
 and an error is thrown in the console:
 
 ```
 chunk-UZ2VXBV6.js?v=b363b4ac:175253 Uncaught (in promise) Error: Unable to resolve resource builtin:/builtin.mylang
    at ResourceModelCollection2.doCreateReferencedObject (chunk-UZ2VXBV6.js?v=b363b4ac:175253:11)
    at ResourceModelCollection2.doCreateReferencedObject (chunk-UZ2VXBV6.js?v=b363b4ac:175251:19)
    at async AsyncReferenceCollection.acquire (chunk-CMPQUMOX.js?v=b363b4ac:2589:26)
    at async TextModelResolverService2.createModelReference (chunk-UZ2VXBV6.js?v=b363b4ac:175357:12)
    at async ReferenceWidget2._revealReference (chunk-UZ2VXBV6.js?v=b363b4ac:94426:17)
```

I was wondering if there is a way to achieve something similar for the web.

**Comment by msujew** [comment]
Hey @dimitrisfasoulas,

you're probably better off asking in a repo that's more dedicated to monaco. Probably at [monaco-languageclient](https://github.com/TypeFox/monaco-languageclient/) or the [monaco repo itself](https://github.com/microsoft/monaco-editor).

cc @kaisalmen
